be/src/storage/lake/spill_mem_table_sink.cpp

```diff
@@ -139,11 +147,102 @@ Status SpillMemTableSink::flush_chunk_with_deletes(const Chunk& upserts, const C
     return Status::OK();
 }
 
+class BlockGroupIterator : public ChunkIterator {
+public:
+    BlockGroupIterator(SpillMemTableSink& sink, Schema schema, const std::vector<spill::BlockPtr>& blocks)
+            : ChunkIterator(schema), _sink(sink), _blocks(blocks) {}
+
+    virtual Status do_get_next(Chunk* chunk) override {
+        if (_block_idx >= _blocks.size()) {
+            return Status::EndOfFile("End of block group");
+        }
+        RETURN_IF_ERROR(_sink.read_chunk_from_group_block(chunk, *_blocks[_block_idx]));
```

Comment:
There will be more than one chunk in one block, need to handle this situation.

Reply:
Another option is change the code which generate the block, and let one block only contains one chunk.

Reply:
fixed